core/exception_vectors.S

```diff
@@ -23,6 +23,13 @@
 #define CAUSE_LOADSTORE 3
 #define CAUSE_LVL1INT 4
 
+        .section .data
+        .balign 4
+        .global debug_exception_handler
+
+debug_exception_handler:
```

Comment:
Can you use a weak-linked symbol for the debug_exception_handler, rather than loading it into RAM?

core/debug_dumps.c might be a good place to define a weak symbol (by default it can be a weak alias to fatal_exception_handler, but if the user links a debug_exception_handler then it can replace this.)